### PR TITLE
add `--state`-parameter and messaging to __package_dpkg

### PR DESCRIPTION
**Pull request by tom-ee**
- `--state` allows to remove a deb-package, if `--purge-if-absent` is
  specified the package is purged instead of "only" removed
- messaging was added
- man-page updated accordingly

### CODE REVIEW

File: cdist/conf/type/__package_dpkg/man.rst [hunk 1]
```diff
@@ -12,30 +12,84 @@ This type is used on Debian and variants (like Ubuntu) to
...
+
+BOOLEAN PARAMETERS
+-------------------
+purge-if-absent
+    use `--purge` instead of just `--remove` for state=absent.
+
+
+BOOLEAN PARAMETERS
+------------------
+purge-if-absent
```
Comment (darko-poljak) [inline cdist/conf/type/__package_dpkg/man.rst]: @tom-ee It seems you have doubled BOOLEAN PARAMETERS section.

File: cdist/conf/type/__package_dpkg/man.rst [hunk 2]
```diff
@@ -12,30 +12,84 @@ This type is used on Debian and variants (like Ubuntu) to
...
+    __package_dpkg baz_1.4_amd64.deb \
+        --source $__type/files/baz_1.4_amd64.deb \
+        --purge-if-absent \
+        --state "absent"
 
 
 SEE ALSO
 --------
 :strong:`cdist-type__package`\ (7)
+:strong:`dpkg-query`\ (1)
```
Comment (darko-poljak) [inline cdist/conf/type/__package_dpkg/man.rst]: @tom-ee List should be comma separated.

**Comment by darko-poljak** [comment]
@tom-ee I assume you have tested all cases?

**Comment by tom-ee** [comment]
Now I did -- and discovered the bug fixed in the already pushed commit.

**Pull request merged**